### ISSUE TITLE
Odd behaviour when (re)assigning themes

### ISSUE THREAD

**Issue by Bloke** [textpattern/textpattern#1456]
### Expected behaviour

Deleting a theme that is assigned to a section as in-development - whether it's in use or not - removes it from the section assignment.

### Actual behaviour

Deleting a theme that is assigned as in-dev is orphaned if it's not used.

### Steps to reproduce

1. Duplicate four-point-eight theme.
2. Rename duplicate as 'second-theme' (title: 'Second theme').
3. Edit Second theme's 'default' page to add "PAGE THEME TWO" to output, and rename it 'page-two'.
4. Assign default section (development only) via Change theme/page/style to: Second theme / page-two / default.
5. Note that theme/page correctly assigned as overrides. Style is common, as the name of the stylesheet is the same in both themes.
6. From the Themes panel, delete second-theme (it's not in use, so this action is permitted).
7. Visit Sections panel. Note that second-theme and page-two are still assigned as dev theme. Clicking to view the front-end page results in 'Unknown section' if dev preview is permitted in prefs.

#### Additional information

Textpattern version: 4.8.0-beta

See #1453.

There are some other strange things going on if Page assets are renamed differently in one theme compared to another. For example behaviour is different if, on the Sections panel, you:

a. Click the Section to edit it.
b. Change theme.
c. Leave the Page dropdown blank.

Result: action permitted.

Compared with:
a. Select Section via multi-edit.
b. Select c'hange theme/page/style'.
c. Change theme.
d. Leave the Page dropdown blank.

Result: action not permitted because the destination Page is not the same as the current page.

Also, some weirdness if a theme has been assigned as dev to a Section and then clicking that section's name in the Sections panel. Changing theme/page assignment can leave blank entries as the 'live' page. Not tracked this down yet, more testing required.

**Comment by bloatware** [comment]
> Textpattern version: 4.8.0-beta

IIRC, we have fixed the first point in #1453. One can not delete an in-dev theme any more.

The second point needs a fix, it's actually a 4.7.3 bug.

**Comment by Bloke** [comment]
Sorry, my bad about the first point. I did get it to misbehave on my 4.8.0-dev, but I think it might have been because I was mucking about with the Page names so it thought it wasn't "in use" any more and allowed me to delete it. I'd have to go back and run the test again to be sure.

There's some oddness about setting a dev theme up, with different Page template names, and then changing the assignment by clicking the Section name in the Sections panel. Like, set a dev and a live theme so both Theme and Page columns show a horizontal line split. Then reassign the Live theme by clicking the Section name and leave the Page blank. I swear I saw it swap the order of the Pages or something. It shows a single Theme but two Pages.

Can't get it to do it now so maybe I imagined it or was using the beta instead of the dev by mistake.
